src/logger/MetricsContext.ts

```diff
@@ -107,14 +108,46 @@ export class MetricsContext {
   }
 
   /**
-   * Validates dimension set length is not more than Constants.MAX_DIMENSION_SET_SIZE
+   * Validates dimension set.
+   * @see [CloudWatch Dimensions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Dimension.html)
    *
    * @param dimensionSet
    */
   public static validateDimensionSet(dimensionSet: Record<string, string>): void {
     if (Object.keys(dimensionSet).length > Constants.MAX_DIMENSION_SET_SIZE)
       throw new DimensionSetExceededError(
-        `Maximum number of dimensions per dimension set allowed are ${Constants.MAX_DIMENSION_SET_SIZE}`)
+        `Maximum number of dimensions per dimension set allowed are ${Constants.MAX_DIMENSION_SET_SIZE}`,
+      );
+
+    // Validate dimension key and value are valid strings
+    Object.entries(dimensionSet).forEach(([key, value]) => {
+      dimensionSet[key] = String(value);
+
+      if (
+        !MetricsContext.isAscii(key) ||
+        key.trim().length == 0 ||
+        key.startsWith(':') ||
+        key.length > Constants.MAX_DIMENSION_NAME_LENGTH
+      ) {
+        throw new InvalidDimensionError(`Dimension value ${value} is invalid`);
+      }
+
+      if (
+        !MetricsContext.isAscii(value) ||
+        value.trim().length == 0 ||
+        value.length > Constants.MAX_DIMENSION_VALUE_LENGTH
+      ) {
+        throw new InvalidDimensionError(`Dimension value ${value} is invalid`);
+      }
```

Comment:
I'm wondering if the error should be more descriptive?

E.g., if the user has a key that is more than 250 characters long with one or more non-ascii characters as well, they would receive the same error message, wouldn't they?

Reply:
What do you suggest I change it to?

Reply:
Just a more specific error message should be fine.

Similar to what GC mentioned here: https://github.com/awslabs/aws-embedded-metrics-node/pull/131#discussion_r950219522